huggingface/hub-docs
# [Widgets] Handle display logic

Closes https://github.com/huggingface/hub-docs/issues/1010

## Description

### This is how it looks like, when an inference api is disabled for a model.

<img width="800" alt="image" src="https://github.com/huggingface/hub-docs/assets/11827707/2774841d-0f48-4cf7-8d38-1c6119c51011">

### This is how it looks like, when an inference api is disabled for a model BUT there is an example with output (#979).
Detailed description:
1. There will be message that says `However, you can still play with [examples that have an output](hf.co/docs/hub/models-widgets#example-outputs)`
2. All the input elements are disabled (you can't enter text, the compute button is greyed out)
3. As a user, you only see examples that have outputs. If in the readme, there was an example without output, it is being filtered out.
4. In the current implementation, when you click an example with an output, it tries to call an inference-api. This should be fixed with #1017

<img width="800" alt="image" src="https://github.com/huggingface/hub-docs/assets/11827707/1360b19b-770a-48fe-8214-1b018c336f83">


### How to test

```
cd js
npm ci
npm run dev -- --open
```
and the command above should open a webpage

### Order of merge operations
1. Merge #1023 
2. Then merge #1063 
3. Then merge #1019 (this PR)
